tests/dynamic-compatibility-test/sonarqube_standard_dynamic_test.go

```diff
@@ -0,0 +1,200 @@
+package tests
+
+import (
+	"crypto/tls"
+	"fmt"
+	"log"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/gruntwork-io/terratest/modules/helm"
+	http_helper "github.com/gruntwork-io/terratest/modules/http-helper"
+	"github.com/gruntwork-io/terratest/modules/k8s"
+
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Teardown resources after test execution: namespace deleted
+func SetupTest(t *testing.T, options *k8s.KubectlOptions, namespaceName string) func(t *testing.T, options *k8s.KubectlOptions, namespaceName string) {
+	log.Println("setup test: Remove leftovers from previous deployments in the same namespace")
+	WaitUntilNamespaceDeletedE(t, options, namespaceName)
+
+	return func(t *testing.T, options *k8s.KubectlOptions, namespaceName string) {
+		log.Println("teardown test")
+		WaitUntilNamespaceDeletedE(t, options, namespaceName)
+	}
+}
+
+// WaitUntilNamespaceDeletedE checks if the `namespaceName` namespace exists and, in the positive case, tries to delete it.
+func WaitUntilNamespaceDeletedE(t *testing.T, options *k8s.KubectlOptions, namespaceName string) {
+	namespace, _ := k8s.GetNamespaceE(t, options, namespaceName)
+	namespaceActualName := namespace.Name
+
+	for namespaceActualName == namespaceName {
+		fmt.Printf("Deleting an existing namespace: %v\n", namespaceActualName)
+		k8s.DeleteNamespaceE(t, options, namespaceActualName) // this is an asyncronous call
+		// verify if the namespace still exists, it takes some time to delete it
+		namespace, _ = k8s.GetNamespaceE(t, options, namespaceName)
+		namespaceActualName = namespace.Name
+		time.Sleep(10 * time.Second)
+	}
+}
+
+// This test checks the dynamic compatibility test for the sonarqube standard and dce charts
+// Specifically, we perform tests that:
+// - check if the required (default) number of pods is running
+// - check if all the running pods have the containersReady condition set to true
+// - check if the sonarqube application is ready (apis are ready to serve new requests)
+func TestSonarQubeChartDynamicCompatibility(t *testing.T) {
+
+	// Input values for our charts' tests
+	table := []struct {
+		name             string
+		chartName        string
+		expectedPods     int
+		values           map[string]string
+		valuesFilesPaths []string
+	}{
+		{"standard-chart-default-values-deployment", "sonarqube", 2, map[string]string{"tests.enabled": "false",
+			//"image.repository":           "sonarqube-arm64-v8", // DO NOT SET: This is only an experimental setting for M1 users
```

Comment:
can we remove the comments from the tests?
I think this should be documented elsewhere, and probably with the work on M1 this won't be needed anymore

Reply:
sure!